src/core/server/server.ts

```diff
@@ -78,6 +80,9 @@ export class Server {
   private readonly coreUsageData: CoreUsageDataService;
   private readonly i18n: I18nService;
 
+  private readonly savedObjectsStartPromise: Promise<SavedObjectsServiceStart>;
+  private resolveSavedObjectsStartPromise!: (value: SavedObjectsServiceStart) => void;
```

Comment:
NIT: we usually prefer `?` instead of `!` for the property declaration and then use `!` when accessing it